nwb_conversion_tools/utils/spike_interface.py

```diff
@@ -607,36 +548,37 @@ def add_electrical_series(
     # channels gains - for RecordingExtractor, these are values to cast traces to uV.
     # For nwb, the conversions (gains) cast the data to Volts.
     # To get traces in Volts we take data*channel_conversion*conversion.
-    channel_conversion = recording.get_channel_gains()
-    channel_offset = recording.get_channel_offsets()
-    unsigned_coercion = channel_offset / channel_conversion
-    if not np.all([x.is_integer() for x in unsigned_coercion]):
-        raise NotImplementedError(
-            "Unable to coerce underlying unsigned data type to signed type, which is currently required for NWB "
-            "Schema v2.2.5! Please specify 'write_scaled=True'."
-        )
-    elif np.any(unsigned_coercion != 0):
-        warnings.warn(
-            "NWB Schema v2.2.5 does not officially support channel offsets. The data will be converted to a signed "
-            "type that does not use offsets."
-        )
-        unsigned_coercion = unsigned_coercion.astype(int)
-    if write_scaled:
-        eseries_kwargs.update(conversion=1e-6)
-    else:
-        if len(np.unique(channel_conversion)) == 1:  # if all gains are equal
-            eseries_kwargs.update(conversion=channel_conversion[0] * 1e-6)
-        else:
+    if recording.has_scaled_traces():
+        channel_conversion = recording.get_channel_gains()
```

Comment:
I have a question.
In my case, the problem is that making a subrecording in `baserecordingextractor` eliminates the gains from the recording object (before passing it to the `write_recording` function). 

Is the goal here to modify the code in a way that we can use `se.SubRecordingExtractor` without having to remove the gains. 

I am failing how this is going to solve the problem described in #330. Or maybe this is somewhere else in the code?

Reply:
As Alessio mentioned, gain propagation through `spikeinterface/spikeinterface` method (`FrameSliceRecording`) seen here: https://github.com/catalystneuro/nwb-conversion-tools/pull/348/files#diff-e1af4b855c44fd131e9ea6bc47af5c62c403c3a044b260362b692ae41dc97a20R51-R53 should in principle resolve the issue naturally without using `SubRecordingExtractors`. So we would also have to replace those subset calls in this PR